File: front/src/app/interfaces/aluno.ts
import { GetTurmaResponse } from "./turma"

export interface CreateAlunoRequest {
  nome: string
  email: string
  telefone?: string
  idTurma: number
}

export interface CreateAlunoResponse {
  id: number
  nome: string
  email: string
  telefone?: string
  turma: GetTurmaResponse
  situacao: string
  ativo: boolean
}

export interface UpdateAlunoRequest {
  nome: string
  email: string
  telefone?: string
  idTurma: number
  situacao: string
}

export interface UpdateAlunoResponse {
  id: number
  nome: string
  email: string
  telefone?: string
  turma: GetTurmaResponse
  situacao: string
  ativo: boolean
}

export interface GetAlunoResponse {
  id: number
  nome: string
  email: string
  telefone?: string
  turma: GetTurmaResponse
  situacao: string
  ativo: boolean
}

export interface AlunoFiltros {
  serie?: number
  turma?: string
  nome?: string
  situacao?: string
  ativo?: boolean
  page?: number
  size?: number
}

File: bibliotech/bibliotech/src/main/java/com/bibliotech/bibliotech/models/Aluno.java
package com.bibliotech.bibliotech.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "aluno")
public class Aluno {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_turma")
    private Turma turma;

    @Column(name = "nome", nullable = false)
    private String nome;

    @Column(name = "email", nullable = false, unique = true)
    private String email;

    @Column(name = "telefone", length = 15)
    private String telefone;

    @Column(name = "ativo", nullable = false)
    private Boolean ativo = true;

    @Column(name = "situacao", length = 20)
    private String situacao = "regular"; // "regular", "debito", "irregular"

    @OneToMany(mappedBy = "idAluno")
    private Set<Emprestimo> emprestimos = new LinkedHashSet<>();

    @PrePersist
    public void prePersist() {
        this.ativo = true;
        this.situacao = "regular";
    }
}